inc/class-tmsc-sync.php

```diff
@@ -0,0 +1,215 @@
+			add_action( 'wp_ajax_get_option_value', array( self::$instance, 'ajax_get_option_value' ) );
+		}
+	}
+
+	/**
+	 * Add CPT and related taxonomies to object menu
+	 */
+	public static function add_menu_pages() {
+		add_submenu_page( 'edit.php?post_type=tms_object', __( 'TMSC Sync', 'tmsc' ), __( 'Sync TMS Objects', 'tmsc' ), 'manage_options', 'tmsc-sync', array( self::$instance, 'render_object_sync_submenu_page' ) );
+	}
+
+	/**
+	 * Render our admin sync template part
+	 */
+	public function render_object_sync_submenu_page() {
+		load_template( TMSCONNECT_PATH . '/templates/tmsc-sync-admin.php' );
+	}
+
+	/**
+	 * Our ajax handler for syncing manually from the wp-admin area submenu.
+	 */
+	public function sync_objects() {
+		if ( current_user_can( self::$capability ) ) {
+			check_ajax_referer( 'tmsc_object_sync', 'tmsc_nonce' );
+			if ( ! empty( $_POST['tmsc-db-host'] ) ) {
+				$host = sanitize_text_field( wp_unslash( $_POST['tmsc-db-host'] ) );
+				update_option( 'tmsc-db-host', $host, false );
+				self::$tms_db_host = $host;
+			}
+			if ( ! empty( $_POST['tmsc-db-name'] ) ) {
+				$name = sanitize_text_field( wp_unslash( $_POST['tmsc-db-name'] ) );
+				update_option( 'tmsc-db-name', $name, false );
+				self::$tms_db_name = $name;
+			}
+			if ( ! empty( $_POST['tmsc-db-user'] ) ) {
+				$user = sanitize_text_field( wp_unslash( $_POST['tmsc-db-user'] ) );
+				update_option( 'tmsc-db-user', $user, false );
+				self::$tms_db_user = $user;
+			}
+			if ( ! empty( $_POST['tmsc-db-password'] ) ) {
+				$password = sanitize_text_field( wp_unslash( $_POST['tmsc-db-password'] ) );
+				update_option( 'tmsc-db-password', $password, false );
+				self::$tms_db_password = $password;
+			}
+			if ( ! empty( $_POST['tmsc-image-url'] ) ) {
+				$url = esc_url_raw( wp_unslash( $_POST['tmsc-image-url'] ) );
+				update_option( 'tmsc-image-url', $url, false );
+				self::$image_url = $url;
+			}
+			/**
+			 * Uncomment the schedule event function and comment the object sync function to enable asynchronous sync.
+			 */
+			self::$instance->object_sync();
+			// wp_schedule_single_event( time(), 'tmsc_cron_events', array() );
+			echo 1;
+		} else {
+			echo 0;
+			wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'tmsc' ) );
+		}
+		exit();
+	}
+
+	/**
+	 * A generic ajax responder that spits back a option value or returns a boolean if a comparison value is passed.
+	 */
+	public function ajax_get_option_value() {
+		check_ajax_referer( 'wp_admin_js_script', 'nonce' );
+		if ( ! empty( $_POST['option_key'] ) ) {
+			// If an old value is passed, return success only if it is different from the current value.
+			// This can be used for asynchrounous checks.
+			$current_value = get_option( sanitize_key( $_POST['option_key'] ) );
+			if ( ! empty( $_POST['old_value'] ) ) {
+				if ( $_POST['old_value'] !== $current_value && __( 'Syncing TMS Objects', 'tmsc' ) !== $current_value ) {
```

Comment:
I do not think you should compare the value of `$current_value` to a translation string `__( 'Syncing TMS Objects', 'tmsc' )` since it is subject to change.  Please make this a static string.